assets/js/autocomplete/suggestions.js

```diff
@@ -133,12 +145,20 @@ function childNodeSuggestion (childNode, parentId, query, category, label) {
 function nodeSectionSuggestion (node, section, query, category, label) {
   if (!matchesAny(section.id, query)) { return null }
 
+  let link
+
+  if (section.anchor === '' || !section.anchor) {
+    link = `${node.id}.html`
+  } else {
+    link = `${node.id}.html#${section.anchor}`
+  }
+
   return {
-    link: `${node.id}.html#${section.anchor}`,
+    link,
     title: highlightMatches(section.id, query),
-    description: node.title,
+    description: section.description || node.title,
     matchQuality: matchQuality(section.id, query),
-    labels: [label, 'section'],
+    labels: section.labels || [label, 'section'],
```

Comment:
This conditional is necessary because we determine the labels from the `type` provided by the search_data item, and it wouldn't make sense for them to have `['extra', 'section']` as labels.